### PR TITLE
Feat: Search Bar Story added

### LINKED ISSUE #1989: [FEATURE] Add page pages/search.js to Storybook
### Description
All React components and pages need to be added to Storybook (in the nextjs branch)

> Note
> Each item listed below for component/page should be in a new Issue and PR

- [ ] pages/search.js

### Screenshots

_No response_

### Additional information

#1944 search component in the nextjs branch for storybook.

### PR DESCRIPTION
**Pull request by ThePiyushAggarwal**
## Fixes Issue
Closes #1989 

## Changes proposed

- [x] Refactored Search Component into a separate file for reuse in story
- [x] Changed the placeholder in Search Bar for better grammar
- [x] Added the Search Bar Story

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4850"></a>

### CODE REVIEW

File: components/SearchBar.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useRef } from "react";
+
+export default function SearchBar({ inputValue, setInputValue }) {
```
Comment (eddiejaoude) [inline components/SearchBar.js:3]: I think this naming would be more clear now it is in a separate component

```suggestion
export default function SearchBar({ value, onChange }) {
```

**Comment by ThePiyushAggarwal** [comment]
I am not able to remove some files from commit.
3 files with .json have gotten inside the PR when I did a merge commit for main into my branch.
I think the pre-commit hooks are running lint commands and making those changes.
As when I tried to manually remove the unnecessary changes, it said commit stopped since it is empty.

**Comment by ChinmayMhatre** [comment]
I can help you with this.

- Go to you fork on github main branch. click the `sync fork` button.
image

- Go to you local repo. use `git checkout main` to navigate to main branch.

- `git fetch upstream`
- `git merge upstream/main`
- `git checkout feature-ThePiyushAggarwal-1989-search-story`

- use these commands to revert the files
  - `git checkout main -- data/AdamsGeeky.json`
  - `git checkout main -- data/SubhadeepZilong.json `
  - `git checkout main --  data/ndzhwr.json `
- `git add . ` (Make sure you don't have any other changes)
- `git commit -m "any message" -n `   (-n will skip the linter)

- `git push origin feature-ThePiyushAggarwal-1989-search-story` push the changes

**Comment by ThePiyushAggarwal** [comment]
Thanks a lot @ChinmayMhatre !

The unnecessary files are not showing up in the PR now.

File: stories/components/SearchBar.stories.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import SearchBar from "../../components/SearchBar";
+
+export default {
+  component: SearchBar,
+  title: "components/Search Bar",
```
Comment (eddiejaoude) [inline stories/components/SearchBar.stories.js:5]: What is `title` for? Other stories don't seen to have this?

File: stories/components/SearchBar.stories.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+};
+
+const Template = (args) => <SearchBar {...args} />;
+
+export const Basic = Template.bind({});
```
Comment (eddiejaoude) [inline stories/components/SearchBar.stories.js:10]: Why `bind` this looks different to our other stories?

File: components/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useRef } from "react";
+
+export default function SearchBar({ value, onChange }) {
```
Comment (eddiejaoude) [inline components/SearchBar.js:3]: I think this should be called `SearchInput`

**Comment by eddiejaoude** [comment]
Thanks @ChinmayMhatre , great collaboration 💪

File: components/SearchBar.js (hunk 4 above)
Comment (ThePiyushAggarwal) [reply components/SearchBar.js]: Done

**Comment by ThePiyushAggarwal** [comment]
I realized there were some errors in the browser console in storybook.

image

Fixed them as well

File: stories/components/SearchBar.stories.js (hunk 2 above)
Comment (ThePiyushAggarwal) [reply stories/components/SearchBar.stories.js]: If I don't use `title`, storybook uses the story file name and it is without a space between "Search" and "Input".

image

File: stories/components/SearchBar.stories.js (hunk 3 above)
Comment (ThePiyushAggarwal) [reply stories/components/SearchBar.stories.js]: This is from the storybook docs. We can easily pass arguments and even reuse those arguments.

image


Let's say we have some arguments for `Basic` component. And we need most of the arguments for `Rounded` component the same as well. We can do this

```
export const Basic = Template.bind({});
Basic.args = {
  color: "red",
};

export const Rounded = Template.bind({});
Rounded.args = {
  ...Basic.args,
  rounded: true,
};
```

**Comment by ThePiyushAggarwal** [comment]
I deleted my fork and this got closed. Anyway, I don't think anyone cares about this PR. Thanks!

**Comment by ChinmayMhatre** [comment]
> I deleted my fork and this got closed. Anyway, I don't think anyone cares about this PR. Thanks!

Hey @ThePiyushAggarwal , I understand the frustration of getting an error. Sometimes PR's and issues get delayed for multiple reasons. 

You see some PR's getting merged quickly as they are simple profile additions and can be reviewed by all the maintainers. Some larger PR are reviewed by specific maintainers and can take some time to be reviewed. All the maintainers have to take time from their work, life etc to review so it does take of time . 

In case the PR's hasn't been reviewed for a week or more which this one is. I would suggest pinging Eddie once, if that is missed you can go to the discord server and put your issue in the `contributor support` channel. Hope you understand.

**Comment by eddiejaoude** [comment]
@ThePiyushAggarwal your PR was great, sorry for being slow in responding, but I get many notifications. Also your PR has `chore: profile addition` because it has profile changes in it, which I leave these PRs for other maintainers and they probably left it for me because it had other code changes - this is why it is so important to keep PRs focused on a single improvement/change.

Yes deleting your fork will close the PR, but you can still get your code changes from the PR if you wish to create a new PR and we can get it merged

Storybook and Playwright automated tests are super important 👍 so I would love this PR if you are happy to open another PR, sorry again for the delay

**Comment by ThePiyushAggarwal** [comment]
Thanks for responding!

I think the `chore: profile addition` lable got added when the linter made some changes in the `.json` files but after that I reverted those.

I am reopening this PR. This PR contains changes for only the storybook for `SearchInput` component.

**Comment by ThePiyushAggarwal** [comment]
Not able to edit this PR to resolve. Closing creating new one.

**Comment by eddiejaoude** [comment]
ok thank you @ThePiyushAggarwal 👍

**Pull request closed without merging**